src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -248,6 +248,34 @@ internal static class UriHelper
             return dest;
         }
 
+        internal static unsafe char[] UnescapeString(string input, int start, int end, char[] dest,
+            ref int destPosition, char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser syntax,
+            bool isQuery)
+        {
+            fixed (char* pStr = input)
+            {
+                return UnescapeString(pStr, start, end, dest, ref destPosition, rsvd1, rsvd2, rsvd3, unescapeMode,
+                    syntax, isQuery);
+            }
+        }
+
+        internal static unsafe char[] UnescapeString(char* pStr, int start, int end, char[] dest, ref int destPosition,
+            char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser syntax, bool isQuery)
+        {
+            PooledCharArray pooledArray = new PooledCharArray(dest);
```

Comment:
Well Stephen said we would have duplicate code to make this method a wrapper around the `UnescapeString` that needs the pooled array. Anyway PooledCharArray is a struct so it will not be allocated in the heap, and therefore we are avoiding `char[]` in this codepath as we are using the PooledCharArray and just copying it over to the `dest char[]` as a lot of APIs that fall in this codepath expect that behavior.